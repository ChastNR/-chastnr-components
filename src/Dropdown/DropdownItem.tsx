import { Link } from 'react-router-dom';

export interface IDropdownItem {
  id: string | number;
  title: React.ReactNode;
  to?: string;
  onClick?: React.MouseEventHandler<HTMLAnchorElement>;
  onKeyDown?: React.KeyboardEventHandler<HTMLAnchorElement>;
}

export const DropdownElement: React.FC<IDropdownItem> = ({ to, title, onClick, onKeyDown }) => (
  <li>
    <Link onClick={onClick} onKeyDown={onKeyDown} to={to}>
      {title}
    </Link>
  </li>
);
