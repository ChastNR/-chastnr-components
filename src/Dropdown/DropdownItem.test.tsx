import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { DropdownElement } from './DropdownItem';

describe('DropdownElement', () => {
  const mockOnClick = jest.fn();
  const mockOnKeyDown = jest.fn();

  it('should render properly', () => {
    const dropdownElement = render(
      <MemoryRouter>
        <DropdownElement
          id="test"
          title="testTitle"
          to="/testUrl"
          onClick={mockOnClick}
          onKeyDown={mockOnKeyDown}
        />
      </MemoryRouter>
    );

    expect(dropdownElement.container.firstChild).toMatchSnapshot();
  });
});
