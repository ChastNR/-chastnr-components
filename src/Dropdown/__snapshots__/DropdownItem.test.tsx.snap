// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownElement should render properly 1`] = `
<li>
  <a
    href="/testUrl"
  >
    testTitle
  </a>
</li>
`;
